src/TestProblem/Hydro/CCSN/Init_TestProb_Hydro_CCSN.cpp

```diff
@@ -201,6 +216,36 @@ void SetParameter()
    if (  ( CCSN_Is_PostBounce == 0 )  &&  ( CCSN_Prob == Post_Bounce )  )
       Aux_Error( ERROR_INFO, "Incorrect parameter %s = %d !!\n", "CCSN_Is_PostBounce", CCSN_Is_PostBounce );
 
+   if ( CCSN_Prob == Core_Collapse )
+   {
+//    CCSN_CC_Red_DT should be smaller than DT__MAX * UNIT_T
+      if (  ( CCSN_Prob == Core_Collapse )  &&  ( CCSN_CC_Red_DT > DT__MAX * UNIT_T )  )
+         Aux_Error( ERROR_INFO, "%s = %13.7e should be smaller than %s = %13.7e !!\n", "CCSN_CC_Red_DT",
+                    CCSN_CC_Red_DT, "DT__MAX * UNIT_T", DT__MAX * UNIT_T );
+
+      if ( CCSN_MaxRefine_FLAG1 ) {
+//       CCSN_MaxRefeine_LV1 should be smaller than MAX_LEVEL
+         if ( CCSN_MaxRefine_LV1 >= MAX_LEVEL )
+            Aux_Error( ERROR_INFO, "%s = %d should be smaller than %s = %d !!\n", "CCSN_MaxRefine_LV1", CCSN_MaxRefine_LV1, "MAX_LEVEL", MAX_LEVEL );
+         if ( CCSN_MaxRefine_FLAG2 ) {
+//          CCSN_MaxRefeine_LV2 should be smaller than MAX_LEVEL
+            if ( CCSN_MaxRefine_LV2 >= MAX_LEVEL )
+               Aux_Error( ERROR_INFO, "%s = %d should be smaller than %s = %d !!\n",        "CCSN_MaxRefine_LV2",   CCSN_MaxRefine_LV2,   "MAX_LEVEL",            MAX_LEVEL            );
+//          CCSN_MaxRefeine_LV1 should be smaller than CCSN_MaxRefeine_LV2
+            if ( CCSN_MaxRefine_LV2 <= CCSN_MaxRefine_LV1 )
+               Aux_Error( ERROR_INFO, "%s = %d should be smaller than %s = %d !!\n",        "CCSN_MaxRefine_LV1",   CCSN_MaxRefine_LV1,   "CCSN_MaxRefine_LV2",   CCSN_MaxRefine_LV2   );
+//          CCSN_MaxRefine_Dens2 should be larger than CCSN_MaxRefine_Dens1
+            if ( CCSN_MaxRefine_Dens2 <= CCSN_MaxRefine_Dens1 )
+               Aux_Error( ERROR_INFO, "%s = %13.7e should be larger than $s = %13.7e !!\n", "CCSN_MaxRefine_Dens2", CCSN_MaxRefine_Dens2, "CCSN_MaxRefine_Dens1", CCSN_MaxRefine_Dens1 ); }}
+//    CCSN_MaxRefeine_LV2 cannot be used solely and CCSN_MaxRefeine_LV1 must be enable first to use CCSN_MaxRefine_FLAG2
+      else if ( CCSN_MaxRefine_FLAG2 ) {
+         Aux_Error( ERROR_INFO, "%s should be enabled first to use %s !!\n", "CCSN_MaxRefine_LV1", "CCSN_MaxRefine_LV2" ); }
```

Comment:
- For `*_LV1/2 >= MAX_LEVEL`, why not check it during loading in `ReadPara()`?
- Why not move the arguments in the following line as the one in line 223?
- Add empty line to make it more readable. Also, move the right } to make the structure more clear.

Reply:
- Now we check this during the loading